crates/hyperqueue/src/dashboard/state.rs

```diff
@@ -1,28 +1,49 @@
 use crate::dashboard::data::DashboardData;
-use crate::dashboard::ui::screen::Screen;
-use crate::dashboard::ui::screens::home::screen::HomeScreen;
+use crate::dashboard::ui::screen::{DashboardScreenState, Screen, ScreenController};
+use crate::dashboard::ui::screens::home::screen::ClusterOverviewScreen;
+use crate::dashboard::ui::screens::worker::screen::WorkerOverviewScreen;
+use std::ops::DerefMut;
 use tako::common::WrappedRcRefCell;
 
 pub struct DashboardState {
-    screens: Vec<Box<dyn Screen>>,
-    current_screen: usize, // TODO: usize newtype wrapper or enum
+    cluster_overview_screen: Box<ClusterOverviewScreen>,
+    worker_info_screen: Box<WorkerOverviewScreen>,
+
     data_source: WrappedRcRefCell<DashboardData>,
+    controller: ScreenController,
 }
 
 impl DashboardState {
     pub fn new(data_source: DashboardData) -> Self {
         Self {
             data_source: WrappedRcRefCell::wrap(data_source),
-            screens: vec![Box::new(HomeScreen::default())],
-            current_screen: 0,
+            cluster_overview_screen: Box::new(ClusterOverviewScreen::default()),
+            worker_info_screen: Box::new(WorkerOverviewScreen::default()),
+            controller: ScreenController::default(),
         }
     }
 
     pub fn get_data_source(&self) -> &WrappedRcRefCell<DashboardData> {
         &self.data_source
     }
 
-    pub fn get_current_screen_mut(&mut self) -> &mut dyn Screen {
-        self.screens[self.current_screen].as_mut()
+    pub fn get_current_screen_and_switcher(&mut self) -> (&mut dyn Screen, &mut ScreenController) {
+        let home_screen = self.cluster_overview_screen.deref_mut();
```

Comment:
When we have screens as separate attributes of the state, and we want to preserve their state, in this function you should match on the enum contained within the controller, and return the corresponding screen.